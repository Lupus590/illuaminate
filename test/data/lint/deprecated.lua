--- @deprecated
local x

--- @deprecated Use something else instead
local y

x = {} -- Assignments to deprecated members are OK.

-- Usages are not.
x.a = 1
print(x, y)
