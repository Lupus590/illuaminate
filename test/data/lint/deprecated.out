deprecated.lua:[10:1-10:1]: Using deprecated member. [var:deprecated]
    │
 10 │ x.a = 1
    │ ^
deprecated.lua:[11:7-11:7]: Using deprecated member. [var:deprecated]
    │
 11 │ print(x, y)
    │       ^
deprecated.lua:[11:10-11:10]: Using deprecated member. [var:deprecated]
Use something else instead
    │
 11 │ print(x, y)
    │          ^
No errors and 3 warnings
